packages/powersync_core/lib/src/sync_status.dart

```diff
@@ -202,3 +215,170 @@ class UploadQueueStats {
     }
   }
 }
+
+@internal
+final class InternalSyncDownloadProgress {
+  final Map<BucketPriority, int> downloaded;
+  final Map<BucketPriority, int> target;
```

Comment:
It may be more accurate if we keep track of the counts for each specific bucket, rather than per priority.

In _most_ cases it shouldn't make a difference, but there are some edge cases where it does:
1. A checkpoint may be "interrupted" before all data is sent. The new checkpoint after the interruption may add buckets, remove buckets, or change the priority of buckets.
2. The counts are not guaranteed to be 100% accurate in all cases. There are some cases such as compacting buckets, which may alter the count without affecting the integrity - potentially while the client is downloading data. To cater for this, we may want to enforce that:
    a. `downloaded <= target` for each bucket (never go over 100%).
    b. `downloaded = target` when we have all data for the bucket.